gridstatus/spp.py

```diff
@@ -193,6 +201,189 @@ def get_load_forecast(self, date, forecast_type="MID_TERM", verbose=False):
+
+        # Don't include the forecast to avoid confusion
+        cols_to_drop = [col for col in df if "Forecast" in col]
+
+        df = df.drop(columns=cols_to_drop + ["Publish Time"]).dropna(
+            subset=["Actual Wind MW", "Actual Solar MW"],
+        )
+
+        if end:
+            return df[df["Interval Start"].between(date, end)]
+
+        return df
+
+    @support_date_range("HOUR_START")
+    def _get_hourly_solar_and_wind_forecast(self, date, end=None, verbose=False):
+        """System-wide wind and solar forecast data for +7days by hour."""
+        url = self._midterm_solar_and_wind_url_from_date(date)
+
+        df = pd.read_csv(url)
+        df = self._process_solar_and_wind_load_or_forecast(
+            df,
+            url,
+            end_time_col="GMTIntervalEnd",
+            interval_duration=pd.Timedelta(hours=1),
+        )
+
+        return df
+
+    @support_date_range("5_MIN")
+    def _get_5_min_solar_and_wind_forecast_and_actuals(
+        self,
+        date,
+        end=None,
+        verbose=False,
+    ):
+        """System-wide wind and solar forecast data for +4 hours in 5-minute intervals.
+        Including actual wind and solar resource totals from previous intervals.
+        """
+        url = self._shortterm_solar_and_wind_url_from_date(date)
+
+        df = pd.read_csv(url)
+        # According to the docs, the end time col should be GMTIntervalEnd, but it's
+        # only GMTInterval in the data
+        df = self._process_solar_and_wind_load_or_forecast(
+            df,
+            url,
+            end_time_col="GMTInterval",
+            interval_duration=pd.Timedelta(minutes=5),
+        )
+
+        return df
+
+    def _process_solar_and_wind_load_or_forecast(
+        self,
+        df,
+        url,
+        end_time_col,
+        interval_duration,
+    ):
+        df = self._handle_market_end_to_interval(
+            df,
+            end_time_col,
+            interval_duration,
+        )
+
+        # Publish time is only an estimate (we'd have to scrape the page to get the
+        # actual publish time)
+        # Generally accurate to within an hour for midterm and 5 mins for shortterm
+        df["Publish Time"] = pd.Timestamp(url.split("-")[-1].split(".")[0])
+
+        df = utils.move_cols_to_front(
+            df,
+            ["Interval Start", "Interval End", "Publish Time"],
+        ).drop(columns=["Time", "Interval"])
+
+        return df.sort_values(["Interval Start", "Publish Time"])
+
+    def _shortterm_solar_and_wind_url_from_date(self, date):
+        hour = date.hour
+        padded_hour = str(hour).zfill(2)
+        padded_hour_plus_one = str((hour + 1) % 24).zfill(2)
+
+        # The first hour in the URL is 1 after the hour in the filename.
+        # Example 2024/01/01/02 has data for 01/01/2024 01:00:00 - 01/01/2024 01:55:00
+        return BASE_SOLAR_AND_WIND_SHORTTERM_URL + date.strftime(
```

Comment:
hmm, not sure i follow why this is necessary, maybe more clear once you simplify these functions

Reply:
The files for intervals 2024-01-01 00:00-00:55 are under the path 2024/01/01/01, the files for 2024-01-01 01:00-01:55 are at the path 2024/01/01/02 etc.